tests/test_staging/test_pg_conversion.py

```diff
@@ -77,3 +77,74 @@ def test_sic_mapper(sic_dummy_data, sic_expected_output, sic_mapper):
     df_result = sic_to_pg_mapper(sic_dummy_data, sic_mapper, target_col="201")
 
     pd.testing.assert_frame_equal(df_result, expected_output_data)
+
+
+@pytest.fixture
+def mapper():
+    mapper_rows = [
+        [36, "N"],
+        [37, "Y"],
+        [45, "AC"],
+        [47, "AD"],
+        [49, "AD"],
+        [50, "AD"],
+        [58, "AH"],
+    ]
+    columns = ["pg_numeric", "pg_alpha"]
+
+    # Create the DataFrame
+    mapper_df = pd.DataFrame(mapper_rows, columns=columns)
+
+    # Return the DataFrame
+    return mapper_df
+
+
+def test_pg_to_pg_mapper_with_many_to_one(mapper):
+    row_data = [["0001", 47, "A"], ["0001", 49, "B"], ["0002", 50, "C"]]
+    columns = ["formtype", "201", "other_col"]
+
+    test_df = pd.DataFrame(row_data, columns=columns)
+
+    expected_data = [["0001", 47, "A", "AD"], ["0001", 49, "B", "AD"]]
+    expected_columns = ["formtype", "201", "other_col", "product_group"]
+    expected_result_df = pd.DataFrame(expected_data, columns=expected_columns)
+    expected_result_df["product_group"] = expected_result_df["product_group"].astype(
+        "category"
+    )
+
+    result = pg_to_pg_mapper(test_df.copy(), mapper.copy())
+    pd.testing.assert_frame_equal(result, expected_result_df, check_dtype=False)
+
+
+def test_pg_to_pg_mapper_success(mapper):
+    row_data = [
+        ["0001", 36, "A"],
+        ["0001", 45, "B"],
+        ["0002", 58, "C"],
+        ["0001", 49, "D"],
+    ]
+    columns = ["formtype", "201", "other_col"]
+
+    test_df = pd.DataFrame(row_data, columns=columns)
+
+    result_df = pg_to_pg_mapper(test_df.copy(), mapper.copy())
+
+    expected_data = [
+        ["0001", 36, "A", "N"],
+        ["0001", 45, "B", "AC"],
+        ["0001", 49, "D", "AD"],
+    ]
+
+    expected_columns = ["formtype", "201", "other_col", "product_group"]
+
+    expected_result_df = pd.DataFrame(
+        expected_data, columns=expected_columns, index=result_df.index
```

Comment:
To the reviewer: please give me your opinion on whether using the index from the result_df is a dodgy move

Reply:
I don't think it's dodgy at all. However, you could also do this:
check_index=False 
or reset index of both dfs, see this stack overflow
https://stackoverflow.com/questions/51655623/how-to-ignore-index-comparison-for-pandas-assert-frame-equal

Reply:
Done